addons/product/models/product_product.py

```diff
@@ -631,8 +631,16 @@ def action_open_documents(self):
     #=== BUSINESS METHODS ===#
 
     def _prepare_sellers(self, params=False):
-        sellers = self.seller_ids.filtered(lambda s: s.partner_id.active and (not s.product_id or s.product_id == self))
-        return sellers.sorted(lambda s: (s.sequence, -s.min_qty, s.price, s.id))
+        company_id = self.env.company.id
+        if params and 'order_id' in params and params['order_id'].company_id:
+            company_id = params['order_id'].company_id.id
+        domain = [
+            ('product_tmpl_id', '=', self.product_tmpl_id.id),
+            ('company_id', '=', company_id),
+        ]
+        sellers = self.sudo().env['product.supplierinfo'].search(domain)
```

Comment:
```suggestion
        sellers = self.sudo().env['product.supplierinfo'].search(
            domain,
            order='sequence asc, min_qty desc, price asc, id asc'
        )
```
Since all the fields we `sort()` on are stored, you could do it with the query (just an option, either way it should be fine)